src/spec_tools/core.cljc

```diff
@@ -130,7 +130,10 @@
         (-options [_] options)
         (-encoder [_ spec _]
           (or (get spec encode-key)
-              (get encoders (parse/type-dispatch-value (:type spec)))
+              (when-let [e (get encoders (parse/type-dispatch-value (:type spec)))]
+                (fn [this x]
+                  (binding [*transformer* nil]
```

Comment:
@miikka if PR #248 is merged, I will need to change this specific code. Would be great to know if this functionality is indeed desired inside spec-tools.